src/Highway.Data.EntityFramework/Factories/DomainRepositoryFactory.cs

```diff
@@ -1,17 +1,14 @@
 ﻿using System;
 using System.Linq;
 
```

Comment:
This winds up looking cluttered in GitHub's diff.  Just view the new file.  The change is pretty simple:

- Added new methods to implement IReadonlyDomainRepositoryFactory (no need to two classes in the case)
- Split RepositoryFactory into its own file.
- Extract a private _CreateRepository_ method to back the non-generic create methods.